artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagingStoreImpl.java

```diff
@@ -289,9 +290,23 @@ public void applySetting(final AddressSettings addressSettings) {
          this.pageLimitBytes = null;
       }
 
+      this.pageLimitMessages = addressSettings.getPageLimitMessages();
```

Comment:
This is in the wrong place, it should be _before_ the matching 'null replacement check' for the value, up on line 273, i.e where it was to begin with (originally as line 271), otherwise it may think the value has changed when it hasnt (but the comparison fouled because the null replacement has been 'bypassed' here).